adapters/envista_il.js

```diff
@@ -0,0 +1,1572 @@
+    }
+
+    let tasks = [];
+    let cityLinks = body.match(/DynamicTable.aspx\?G_ID=(\d*)/g);
+    while (cityLinks.length > 0) {
+      let link = source.url + cityLinks.pop();
+      tasks.push(handleCity(source, link));
+    }
+
+    parallelLimit(tasks, 2, (err, results) => {
+      if (err) {
+        return cb(err, []);
+      }
+
+      results = flattenDeep(results);
+      results = convertUnits(results);
+
+      return cb(err, {name: 'unused', measurements: results});
+    });
+  });
+}
+
+const handleCity = function (source, link) {
+  return function (done) {
+    request(link, (err, res, body) => {
+      if (err || res.statusCode !== 200) {
+        return done(null, []);
+      }
+      let tasks = [];
+      let stationIds = body.match(/StationInfo[5]?.aspx\?ST_ID=(\d*)/g);
+      let $ = cheerio.load(body);
+      let regionName = $('#lblCaption').text().split(' - ')[1];
+      while (stationIds.length > 0) {
+        let link = source.url + stationIds.pop();
+        link = link.replace(/StationInfo[5]?/, 'StationReportFast');
+        tasks.push(handleStation(source, link, regionName));
+      }
+      parallel(tasks, (err, results) => {
+        return done(err, results);
+      });
+    });
+  };
+};
+
+// station query page
+const handleStation = function (source, link, regionName) {
+  return function (done) {
+    request(link, (err, res, body) => {
+      if (err || res.statusCode !== 200) {
+        return done(null, []);
+      }
+
+      const $ = cheerio.load(body);
+      if (!hasAcceptedParameters($)) {
+        return done(null, []);
+      }
+
+      // Form inputs
+      let form = {};
+      form['__VIEWSTATE'] = $('#__VIEWSTATE').attr('value');
+      form['__VIEWSTATEGENERATOR'] = $('#__VIEWSTATEGENERATOR').attr('value');
+      let lstMonitors = encodeURIComponent(getLstMonitors($));
+      // the system kept the / symbol unencoded
+      lstMonitors = lstMonitors.replace(/%2F/g, '/');
+      form['lstMonitors'] = lstMonitors;
+      form['chkall'] = 'on';
+      form['RadioButtonList1'] = 0;
+      form['RadioButtonList2'] = 0;
+
+      // base for txtStartTime_p & txtEndTime_p
+      const txtEndTime = moment(
+        $('#BasicDatePicker2_TextBox').attr('value'),
+        'DD-MM-YYYY'
+      );
+
+      form['BasicDatePicker1$textBox'] = $('#BasicDatePicker1_TextBox').attr('value');
+      form['txtStartTime'] = '00:00';
+      form['BasicDatePicker2$textBox'] = $('#BasicDatePicker2_TextBox').attr('value');
+      form['txtEndTime'] = '00:00';
+      form['txtEndTime_p'] = txtEndTime.format('YYYY-M-DD') + '-0-0-0-0';
+      form['txtStartTime_p'] = txtEndTime.subtract(1, 'day').format('YYYY-M-DD') + '-0-0-0-0';
+      form['ddlAvgType'] = 'AVG';
+      form['ddlTimeBase'] = 15;
+      form['btnGenerateReport'] = 'הצג+דוח';
+
+      // used to ignore fallbacks w/ 5 min avaraging period
+      const minAvgPeriod = parseInt($('#ddlTimeBase').children()[0].attribs.value);
```

Comment:
What's getting captured here is the minimum averaging period provided by the source for a given station. On the site you are able to select from multiple averaging periods for a given station. 

This min averaging period matches the averaging period reported on the page I was gathering the station fallback data. So the idea was if the averaging period for a given station was less than 10 minutes, the adapter would ignore trying to call the fallback.